### PR TITLE
Refactor code into smtp package

### PR DESCRIPTION
**Pull request by JohannesEbke**
@MartinAltmayer to review

### CODE REVIEW

File: smtp/send.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+// SendMailer interface is for classes that can send mail.
+type SendMailer interface {
+	SendMail(MailEnvelope) error
+}
+
+// Send the envelope via the specified SendMailer
+func (envelope MailEnvelope) Send(mailer SendMailer) (err error) {
+	return mailer.SendMail(envelope)
+}
```
Comment (MartinAltmayer) [inline smtp/send.go]: Instead of using this method, one should use mail.SendMail directly.

File: smtp/send.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	if err = c.Rcpt(envelope.Recipient); err != nil {
+		return err
+	}
+
+	// Send the email body.
+	wc, err := c.Data()
+	if err != nil {
+		return err
+	}
+	_, err = fmt.Fprintf(wc, envelope.Message)
```
Comment (MartinAltmayer) [inline smtp/send.go]: Fprint?

File: smtp/send.go (hunk 1 above)
Comment (JohannesEbke) [inline smtp/send.go]: Done.

File: smtp/send.go (hunk 2 above)
Comment (JohannesEbke) [inline smtp/send.go]: Replaced by direct Write()

**Pull request merged**